backend/src/main/java/com/nikogrid/backend/controllers/AuthenticationController.java

```diff
@@ -0,0 +1,102 @@
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.server.Cookie.SameSite;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/v1/auth")
+@Validated
+public class AuthenticationController {
+    private final AuthenticationService authenticationService;
+    private final AuthenticationManager authenticationManager;
+    private final JwtGenerator jwtGenerator;
+    private final int jwtExpirationSeconds;
+
+    @Autowired
+    public AuthenticationController(
+            AuthenticationService authenticationService,
+            AuthenticationManager authenticationManager,
+            JwtGenerator jwtGenerator,
+            @Value("${jwt.expiration-seconds}") int jwtExpirationSeconds
+    ) {
+        this.authenticationService = authenticationService;
+        this.authenticationManager = authenticationManager;
+        this.jwtGenerator = jwtGenerator;
+        this.jwtExpirationSeconds = jwtExpirationSeconds;
+    }
+
+    @PostMapping("/login")
+    @PreAuthorize("!isAuthenticated()")
+    @ResponseStatus(HttpStatus.OK)
+    public AuthResponse authenticateUser(HttpServletRequest request, HttpServletResponse response, @Valid @RequestBody LoginDTO loginDTO) {
+        final var authenticationRequest = UsernamePasswordAuthenticationToken
+                .unauthenticated(loginDTO.getEmail(), loginDTO.getPassword());
+        final var authentication = this.authenticationManager.authenticate(authenticationRequest);
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+
+        if (authentication == null) {
+            throw new BadCredentialsException("Invalid Credentials");
+        }
+
+        final var token = this.jwtGenerator.generateToken(authentication);
+
+        final var cookie = new Cookie("AUTH", token);
+        cookie.setMaxAge(jwtExpirationSeconds);
+        cookie.setPath("/");
+        cookie.setSecure(request.isSecure());
```

Comment:
This won't work when you are behind a reverse proxy, you need to make this configurable.